src/config/field-overrides/program-rules/programRuleActionDialog.component.js

```diff
@@ -237,36 +331,113 @@ class ProgramRuleActionDialog extends React.Component {
         });
     }
 
+    async validate(expression) {
+        let d2
+        let api
+
+        if (!this.props.program || !this.props.program.id) {
+            return;
+        }
 
+        try {
+            d2 = await getInstance();
+            api = d2.Api.getApi();
+        } catch (error) {
+            return
+        }
+
+        const pendingStatus = {
+            status: ExpressionStatus.PENDING,
+            message: d2.i18n.getTranslation('checking_expression_status'),
+        }
+        this.setState(prevState => ({
+            ...prevState,
+            status: pendingStatus
+        }))
+
+        const programId = this.props.program.id;
+        const url = `programRuleActions/data/expression/description?programId=${programId}`;
+        const requestOptions = {
+            headers: {
+                'Content-Type': 'text/plain',
+            },
+        };
+
+        try {
+            const { status, description, message } = await api.post(
+                url,
+                `"${expression}"`,
+                requestOptions
+            );
+            const isOk = status === 'OK';
+
+            const newStatus = {
+                status: isOk
+                    ? ExpressionStatus.VALID
+                    : ExpressionStatus.INVALID,
+                message: isOk ? description : message,
+                details: description,
+            };
+
+            this.setState(prevState => ({
+                ...prevState,
+                status: newStatus,
+            }));
+        } catch (error) {
+            // If error contains a message and an error status we consider it to be a valid response
+            if (error.message && error.status === 'ERROR') {
+                const newStatus = {
+                    status: ExpressionStatus.INVALID,
+                    message: error.message,
+                }
+
+                this.setState(prevState => ({
+                    ...prevState,
+                    status: newStatus,
+                }));
+            }
+
+            // Rethrow if not a valid error
+            throw error;
```

Comment:
I think we should gracefully handle this as well, `api.post` will throw on network error, or eg. 500 errors. I think we should inform the user of that instead of rethrowing.